src/App.tsx

```diff
@@ -17,18 +57,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterOption={filterOption}
+                setFilterOption={setFilterOption}
+                setQuery={setQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0 ? (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodo={selectedTodo}
+                  showTodo={(todo) => {
+                    setSelectedTodo(todo);
+                  }}
+                />
+              ) : (
+                <Loader />
```

Comment:
I would create other state like `isLoading ` and show `<Loader />` only when `isLoading  = true`. To show loading when it is really needed